### DISCUSSION TITLE
Custom command name

### DISCUSSION THREAD

**Discussion by xiao-e-yun** [tauri-apps/tauri#3620] [Q&A]
I want to separate the code, but this results in ambiguous command names
How to customize the name

`main.rs`
```rust
mod other;

fn main() {
  tauri::Builder::default()
    .invoke_handler(tauri::generate_handler![
      other::hello,
    ])
    .run(tauri::generate_context!())
    .expect("error while running tauri application");
}
```

`other.rs`
```rust
#[tauri::command]
pub fn hello() -> i32 {
  1
}
```

`main.js`
```js
__TAURI_INVOKE__("hello") // 1
```

**Comment by JonasKruckenberg** [comment]
In it's current implementation the command name is tied to the function name. So to rename the command, change `fn hello` to `fn hello_world` or something

**Reply by xiao-e-yun** [reply]
belike `mod1::mod2::mod1_mod2_function`?
or
```rust
pub fn hello{
  println!("hello");
}

#[tauri::command]
pub fn mod1_hello(){
  hello();
}
```

I know, but I'm hoping for a better solution

**Reply by JonasKruckenberg** [reply]
Well, chances are you can name your functions something more specific that avoids clashes with other modules. At least that's my experience working with large Tauri codebases.

But we will be reworking the IPC system in the future, addressing this issue in the process.

**Reply by SuiltaPico** [reply]
> Well, chances are you can name your functions something more specific that avoids clashes with other modules. At least that's my experience working with large Tauri codebases.
> 
> But we will be reworking the IPC system in the future, addressing this issue in the process.

Will this problem be solved in v2?

**Comment by ThunbergOlle** [comment]
Any updates on this?
